### PR TITLE
feat: update cli-utils to v0.31.1

### PR DESCRIPTION
**Pull request by karlkfi**
- Replace StatusPoller w/ StatusWatcher.
  Live commands are now be faster and more responsive.
- Remove obsolete --poll-period command option
- Replace forked cli-utils with upstream version
- Error on Skip, with clear error message, instead of skipping silently
- Combine Applied, Changed, Unchanged, and ServersideApplied statuses
  into one ApplySuccessful status
- Use Successful status when there's no diff, and Skipped when
  apply is filtered, instead of Unchanged for both.
- Replace apply/prune/delete Unspecified with Pending
- Add Failed status to use when there is an Error (except when Skipped)
- Improve readability and consistency of event enum strings
- Print event summary after each phase
- Modify JSON printer to match new event enums
- Fix flaky stdout tests by filtering optional "reconcile pending" output
  and client-side throttling log messages

### CODE REVIEW

**Comment by karlkfi** [comment]
looks like `end-to-end-test.sh` is hanging (never reconciling?) and the outputs need to be updated.

**Comment by karlkfi** [comment]
lol... nvm. It's hung up on `Please enter Username:`

I couldn't figure it out until I switched it to up `tee` to print and capture stdout/stderr.

**Comment by karlkfi** [comment]
Blocked on https://github.com/GoogleContainerTools/kpt/pull/3166

**Comment by karlkfi** [comment]
Rebased. Hopefully the e2e tests fail now instead of hanging forever on the password prompt.

Still need to update all the expected output in the bash tests. I wish they tested the full output instead of just individual lines...

**Comment by karlkfi** [comment]
Found a bug in the new StatusWatcher. This causes the e2e tests to hang until they hit the 72 hour timeout.
Fix: https://github.com/kubernetes-sigs/cli-utils/pull/585

Found another bug, but I'm not sure if it's new. Apply statuses seem to not be ordered the way I thought they would be (alphabetical by name). So there's an e2e test where a failure and success are being randomly re-ordered. If it's not a new bug, I may just change that test back to spot checking lines, instead of checking the whole output.

**Review by mengqiy — APPROVED** [review]
Thanks for cleaning up the stuff under `thirdparty`.

**Comment by karlkfi** [comment]
Given the significant changes to kpt live behavior & output, I'd like to get one more review from @mortent and/or @droot before merging.

File: pkg/test/live/runner.go
Comment (karlkfi) [reply pkg/test/live/runner.go:239]: Good question. I think I copied the for loop from one that used a switch and needed to break out of the loop and not just the switch. I'll clean this up.

**Review by mortent — APPROVED** [review]
Looks good to me.

File: pkg/test/live/config.go
Comment (karlkfi) [reply pkg/test/live/config.go:39]: The why is that the `reconcile pending` lines are sometimes printed and sometimes not, depending on timing. This config field tho is generic and could be used for other use cases for filtering optional output lines.

Got a suggestion for better comment?

Comment (droot) [reply pkg/test/live/config.go:39]: Ah..now makes sense :) Since this is a test config, wouldn't worry too much.

File: pkg/test/live/runner.go
Comment (droot) [reply pkg/test/live/runner.go:239]: Aah. I kept staring at the code for sometimes wondering what am I missing here :)

**Review by droot — APPROVED** [review]
Looks good (except the minor comment about `continue`). Feel free to merge this after cleaning that up.

File: pkg/test/live/runner.go
Comment (karlkfi) [reply pkg/test/live/runner.go]: Fixed. I also fixed the other two filter functions to not use the confusing loop label pattern.

**Pull request merged**
